healthcheck.go

```diff
@@ -53,6 +55,71 @@ type ClusterHealth struct {
 	Status string `json:"status"`
 }
 
+//indexcheck calls elasticsearch to check if an index from the client exists
+//What is a context?
+func (cli *Client) indexcheck(ctx context.Context) (code int, err error) {
+
+	var clientElasticSearchIndices = cli.indices
+
+	for _, index := range clientElasticSearchIndices {
```

Comment:
We can gain a small performance improvement here by removing the extra variable assignment.  Accessing the `indices` field is a quick get operation so it is safe to use it directly in the for range. The variable assignment used here is only necessary if there was significant computation required to get the value and that value was used multiple times as then we would be saving the extra computation.

```suggestion
	for _, index := range cli.indices {
```